content/docs/faq-functions.md

```diff
@@ -0,0 +1,171 @@
+  handleClick = () => {
+    console.log('Click happened')
+  }
+  render() {
+    return <button onClick={this.handleClick}>Click Me</button>
+  }
+}
+```
+
+#### Bind in Render
+
+```jsx
+class Foo extends Component {
+  handleClick () {
+    console.log('Click happened')
+  }
+  render() {
+    return <button onClick={this.handleClick.bind(this)}>Click Me</button>
+  }
+}
+```
+
+#### Arrow Function in Render
+
+```jsx
+class Foo extends Component {
+  handleClick () {
+    console.log('Click happened')
+  }
+  render() {
+    return <button onClick={() => this.handleClick()}>Click Me</button>
+  }
+}
+```
+
+**Note**: Using an arrow function or binding in render creates a new function each time the component renders, which may have performance implications (see below).
+
+### Is it OK to use arrow functions in render methods?
+
+Generally speaking, yes, it is OK, and it is often the easiest way to pass parameters to callback functions.
+
+If you do have performance issues, by all means, optimize!
+
+### Why is my function being called every time the component re-renders?
+
+Make sure you aren't _calling the function_ when you pass it to the component:
+
+```jsx
+render() {
+  {/* handleClick is called instead of passed as a reference! */}
+  return <button onClick={this.handleClick()}>Click Me</button> 
+}
+```
+
+### How do I pass a parameter to an event handler or callback?
+
+You can use an arrow function to wrap around an event handler and pass parameters: 
+
+```jsx
+<Element onClick={() => this.handleClick(id)} />
+```
+
+This is equivalent to calling `.bind`:
+
+```jsx
+<Element onClick={this.handleClick.bind(this, id)} />
+```
+
+#### Example: Passing params using arrow functions
+
+```jsx
+class Component extends React.Component {
+  state = {
+      justClicked: 0,
+      items: Array.from({length: 5}, (_, i) => i)
+  }
+  
+  handleClick = i => this.setState({ justClicked: i })
+  
+  render () {
+    return (
+      <div>
+        Just clicked: {this.state.justClicked}
+        <ul>
+          { this.state.items.map(i => 
+            <li onClick={() => this.handleClick(i)}>
+              Item: {i}
+            </li>
+          ) }
+        </ul>
+      </div>
+   )
+  }
+}
+```
+
+#### Example: Passing params using data-attributes
+
+Alternately, you can use DOM APIs to store data needed for event handlers. Consider this approach if you need to optimize a large number of elements or have a render tree that relies on React.PureComponent equality checks.
+
+```jsx
+class Component extends React.Component {
+  state = {
+      justClicked: 0,
+      items: Array.from({length: 5}, (_, i) => i)
+  }
+  
+  handleClick = event => {
+    this.setState({
+      justClicked: event.target.dataset.i
+    })
+  }
+  
+  render () {
+    return (
+      <div>
+        Just clicked: {this.state.justClicked}
+        <ul>
+          { this.state.items.map(i => 
+            <li data-i={i} onClick={this.handleClick}>
```

Comment:
Add missing `key` attribute.

Reply:
Since I generated the Array in the constructor, what do you suggest for the key? I could change the example to assume it has some other unique attributes I suppose.

Reply:
That would be fine. I just don't think we should show an example that omits an important, required property like `key`.